### PR TITLE
Reimplemented edit advice method

### PR DESCRIPTION
**Pull request by AdrYne01**

### CODE REVIEW

File: app/controllers/advice_controller.rb [hunk 1]
```diff
@@ -1,70 +1,86 @@
...
+      if invalid_advice?(sorted_advice, num_advices, question)
+        # The number of advices for this question has changed.
+        QuestionnaireHelper.adjust_advice_size(@questionnaire, question)
+      end
+    end
+  end
+
+  # Function to calculate number of advices for the current question attribute based on max and min question score.
+  # Method name is consistent with the functionality
+  def calculate_num_advices(question)
```
Comment (bdevine2) [inline app/controllers/advice_controller.rb:54]: I think you could simplify this into more modern grammar.  Something like question.is_a?(ScoreQuestion) ? <true leg> : <false leg>

**Review by bdevine2 — CHANGES_REQUESTED** [review]
I suggested one small change but the rest looks good

File: app/controllers/advice_controller.rb (hunk 1 above)
Comment (AdrYne01) [reply app/controllers/advice_controller.rb:54]: **if question.is_a?(ScoredQuestion)** This line checks if the current question attribute is scored or not by checking if it is of the type 'ScoredQuestion'. Don't you think 'ScoredQuestion' makes more sense in that case?

Comment (bdevine2) [reply app/controllers/advice_controller.rb:54]: What I meant exactly was changing it to this

question.is_a?(ScoredQuestion) ? @questionnaire.max_question_score - @questionnaire.min_question_score + 1 : 0

It feels much more modern and elegant.

Comment (AdrYne01) [reply app/controllers/advice_controller.rb:54]: Oh okay, I understand now. My bad. I will make the necessary changes.

Comment (AdrYne01) [reply app/controllers/advice_controller.rb:54]: I have updated the file.

**Review by bdevine2 — APPROVED** [review]

**Review by aebartl3 — COMMENTED** [review]
lgtm

**Pull request merged**